### PR TITLE
Refactor Config APIs

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

**Comment by joaoandre-avaiga** [comment]
Please review the text on the docstrings as well. Let me know any sugestions.

File: src/taipy/config/config.py [hunk 1]
```diff
@@ -58,13 +58,13 @@ def global_config(cls) -> GlobalAppConfig:
     @classmethod
     @_ConfigBlocker._check()
     def load(cls, filename):
-        """Load a configuration file.
+        """Load a configuration file from Taipy Studio and replaces the current python config.
```
Comment (jrobinAV) [inline src/taipy/config/config.py]:
```suggestion
        """Load a configuration file to replace the current python config and trigger the Config compilation.
```

Comment (jrobinAV) [inline src/taipy/config/config.py]:
```suggestion
        """Load a configuration file to replace the current python config and trigger the Config compilation.
```

File: src/taipy/config/config.pyi [hunk 2]
```diff
@@ -25,27 +25,45 @@ class Config:
...
+
     @_Classproperty
     def global_config(cls) -> GlobalAppConfig:
         """Return configuration values related to the global application as a `GlobalAppConfig^`."""
+
     @classmethod
     @_ConfigBlocker._check()
     def load(cls, filename):
-        """Load a configuration file.
+        """Load a configuration file from Taipy Studio and replaces the current python config.
```
Comment (jrobinAV) [inline src/taipy/config/config.pyi]:
```suggestion
        """Load a configuration file from Taipy Studio and replaces the current python config and trigger the Config compilation.
```

**Review by jrobinAV — COMMENTED** [review]
I believe the restore method is missing.

It should replace the applied config with the loaded toml file. It should NOT trigger recompilation.

**Review by trgiangdo — APPROVED** [review]
The .pyi file is looking good.

Imo, the docstring may be confusing to the user.
But since it's in the Dev manual, a good User manual should be sufficient.

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**